src/core/system.cpp

```diff
@@ -367,13 +400,20 @@ Eigen::MatrixXd velocity_at_targets(MatrixRef &r_trg) {
     const auto [sol_fibers, sol_shell, sol_bodies] = get_solution_maps(curr_solution_.data());
     const auto &fp = params_.fiber_periphery_interaction;
 
-    Eigen::MatrixXd f_on_fibers = fc_.apply_fiber_force(sol_fibers);
-    if (params_.periphery_interaction_flag) {
-        int i_fib = 0;
-        for (const auto &fib : fc_) {
-            f_on_fibers.col(i_fib) += shell_->fiber_interaction(fib, fp);
-            i_fib++;
+    Eigen::MatrixXd f_on_fibers = fc_->apply_fiber_force(sol_fibers);
+    // XXX FIXME This touches the fibers directly, don't do that
```

Comment:
Maybe we should just have pointers for each of the different implementations available, or something like
`FiberContainerFiniteDifference *system::get_fc_fd() { return noisy_cast_nonsense; }` 

This maybe should also be 'simplified' and passed down to the periphery object, with overloading to have `fiber_interaction_finitediff() -> fiber_interaction()`. 

The best approach is probably just making a `FiberContainer::get_forces_from_periphery` routine or something along those lines, as the comment suggests.